src/hotspot/share/services/cpuTimeUsage.cpp

```diff
@@ -76,6 +98,24 @@ jlong CPUTimeUsage::GC::stringdedup() {
   return 0;
 }
 
+CPUTime CPUTimeUsage::GC::detailed_gc_threads() {
+  DetailedCPUTimeThreadClosure cl;
+  Universe::heap()->gc_threads_do(&cl);
+  return cl.cpu_time();
+}
+
+CPUTime CPUTimeUsage::GC::detailed_gc_operation_vm_thread() {
+  assert_at_safepoint();
+  return detailed_thread_cpu_time_or_zero((Thread*)VMThread::vm_thread());
+}
+
+CPUTime CPUTimeUsage::GC::detailed_stringdedup() {
+  if (UseStringDeduplication) {
+    return detailed_thread_cpu_time_or_zero((Thread*)StringDedup::_processor->_thread);
+  }
+  return { 0, 0 };
```

Comment:
I would have turned this around, but that's just taste, so skip if you prefer as is: 
```suggestion
CPUTime CPUTimeUsage::GC::detailed_stringdedup() {
  if (!UseStringDeduplication) {
    return {0, 0};
  }
  return detailed_thread_cpu_time_or_zero((Thread*)StringDedup::_processor->_thread);
```